Fix https://github.com/erincatto/box2d/pull/609/files

Kira-PH opened the pull request:
Polygon updated as per https://github.com/erincatto/box2d/pull/609/files
Sweep updated as per https://github.com/erincatto/box2d/pull/610/files
Static body flags updated as per https://github.com/erincatto/box2d/pull/614/files
Would be good to copy Erin's unit tests from those commits and get his documentation addition in our documentation

Commit:
Fix https://github.com/erincatto/box2d/pull/609/files

Commit:
Fix https://github.com/erincatto/box2d/pull/610/files

Commit:
Fix https://github.com/erincatto/box2d/pull/610/files

Commit:
Fix https://github.com/erincatto/box2d/pull/614/files
Removed unused fields
Fixed a bug when removing a fixture from a body
Removed nullability because it doesn't really add anything at this point
Added equality stuff for Vec2
Removed some Vec2 warnings
Made DebugDraw methods obsolete

Commit:
Moved all classes/structs to their own files
Updated joints as per https://github.com/erincatto/box2d/pull/618/files
Made SetOneSided public

Commit:
Added helpful ObsoleteAttributes and extended Initialize methods for JointDefs losing Frequency and DampingRatio

benzuk commented:
> Polygon updated as per https://github.com/erincatto/box2d/pull/609/files
> Sweep updated as per https://github.com/erincatto/box2d/pull/610/files
> Static body flags updated as per https://github.com/erincatto/box2d/pull/614/files
> Would be good to copy Erin's unit tests from those commits and get his documentation addition in our documentation

Hey dude, sorry for not being available last week. I just finished my military service and I had a lot of work. I will check these changes ASAP.

In the coming days, I will make the code more readable (no unit tests now and no logic changes). Something similar in my recent PRs.

Commit:
Fixed a bug in RevoluteJoint where upper limit was enforced even when enableLimit was set to false.
Changed frequencyHz and dampingRatio to be advisory obsolete, but also added implicit conversion to stiffness & damping

HughPH commented:
No problem! Glad to hear you finished your military service, must be a relief for you!
